### PR TITLE
feat: Remove React Router from Menu

### PR DESCRIPTION
**Pull request by jeffredodd**
### Description
In order to remove the dependency on React Router from Fundamentals-React (#115), we must refactor the the following components...

- Breadcrumb (#294)
- Menu (#293)
- Tabs (Coming Soon)
- SideNavigation (Coming Soon)

This PR is focused on **Menu** and does introduce breaking changes such as...

- Removal of Link prop
- Removal of LinkProps prop

**React Router Menu Example (NEW)**
```
<Menu>
    <MenuList>
        <MenuItem>
            <Link to='/'>Option 1</Link>
        </MenuItem>
        <MenuItem>
            <Link to='/'>Option 2</Link>
        </MenuItem>
        <MenuItem>
            <Link to='/'>Option 3</Link>
        </MenuItem>
    </MenuList>
</Menu>
```

**Anchor via children (NEW)**
```
<Menu>
    <MenuList>
        <MenuItem>
            <a href='/'>Option 1</a>
        </MenuItem>
        <MenuItem>
            <a href='/'>Option 2</a>
        </MenuItem>
        <MenuItem>
            <a href='/'>Option 3</a>
        </MenuItem>
    </MenuList>
</Menu>
```

**Anchor via Props (Classic)**
```
<Menu>
    <MenuList>
        <MenuItem url="/">Option 1</MenuItem>
        <MenuItem url="/">Option 2</MenuItem>
        <MenuItem url="/">Option 3</MenuItem>
        <MenuItem url="/">Option 4</MenuItem>
    </MenuList>
</Menu>
```

### CODE REVIEW

File: src/Menu/Menu.Component.js [hunk 1]
```diff
@@ -82,11 +105,7 @@ export const MenuComponent = () => {
                         name: 'addonProps',
                         description: 'object - additional props to be spread to the addon section'
                     },
-                    { name: 'link', description: 'string - a router link. Use either \'url\' or \'link\'' },
-                    {
-                        name: 'linkProps',
-                        description: 'object - additional props to be spread to the Menu Item links'
-                    },
+                    { name: 'children', description: 'component - can be used to pass React Router <Link> or any other component which emits an <a>.' },
```
Comment (jbadan) [inline src/Menu/Menu.Component.js]: Be careful if https://github.com/SAP/fundamental-react/pull/291 gets merged first, you'll need to move the description into the new object.

File: src/Menu/Menu.test.js [hunk 2]
```diff
@@ -157,7 +187,9 @@ describe('<Menu />', () => {
                 <MemoryRouter>
                     <Menu>
                         <MenuList>
-                            <MenuItem link='/' linkProps={{ 'data-sample': 'Sample' }} />
+                            <MenuItem urlProps={{ 'data-sample': 'Sample' }}>
+                                <Link to='/' />
+                            </MenuItem>
                         </MenuList>
```
Comment (jbadan) [inline src/Menu/Menu.test.js]: Are this and the next test now testing the same thing?

File: src/Menu/Menu.Component.js (hunk 1 above)
Comment (greg-a-smith) [reply src/Menu/Menu.Component.js]: It's been merged so you can be the first new consumer of the updated prop documentation. 👍

Comment (jeffredodd) [reply src/Menu/Menu.Component.js]: Fixed!

File: src/Menu/Menu.test.js (hunk 2 above)
Comment (jeffredodd) [reply src/Menu/Menu.test.js]: They are not actually. One is testing using the React-Router `<Link>` where the other is testing using the internal creation of the `<a>` tag.

File: src/Menu/Menu.js [hunk 3]
```diff
@@ -45,34 +44,42 @@ MenuList.propTypes = {
 };
 
 // ---------------------------------------- Menu Item ----------------------------------------
-export const MenuItem = ({ url, link, isLink, separator, addon, children, onclick, className, addonProps, linkProps, urlProps, ...props }) => {
+export const MenuItem = ({ url, link, isLink, separator, addon, children, onclick, className, addonProps, urlProps, ...props }) => {
```
Comment (greg-a-smith) [inline src/Menu/Menu.js]: The `link` prop type was removed so can it be removed from here as well?

**Comment by jeffredodd** [comment]
Fixed the issue with the build. Simply had to make sure when spreading our `className` we merged with consumer provided classNames as well. If anyone doesn't object i'll merge this shortly.

**Pull request merged**